File: src/main/java/net/atos/zac/app/zaken/converter/RESTZaaktypeRelatieConverter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaken.converter;

import java.net.URI;

import net.atos.client.zgw.ztc.model.generated.ZaakTypenRelatie;
import net.atos.zac.app.zaken.model.RESTZaaktypeRelatie;
import net.atos.zac.app.zaken.model.RelatieType;
import net.atos.zac.util.UriUtil;

public class RESTZaaktypeRelatieConverter {

  public RESTZaaktypeRelatie convertToRESTZaaktypeRelatie(
      final URI zaaktypUri, final RelatieType relatieType) {
    final RESTZaaktypeRelatie restZaaktypeRelatie = new RESTZaaktypeRelatie();
    restZaaktypeRelatie.zaaktypeUuid = UriUtil.uuidFromURI(zaaktypUri);
    restZaaktypeRelatie.relatieType = relatieType;
    return restZaaktypeRelatie;
  }

  public RESTZaaktypeRelatie convertToRESTZaaktypeRelatie(final ZaakTypenRelatie zaakTypenRelatie) {
    final RESTZaaktypeRelatie restZaaktypeRelatie = new RESTZaaktypeRelatie();
    restZaaktypeRelatie.zaaktypeUuid = UriUtil.uuidFromURI(zaakTypenRelatie.getZaaktype());
    restZaaktypeRelatie.relatieType = RelatieType.valueOf(zaakTypenRelatie.getAardRelatie().name());
    return restZaaktypeRelatie;
  }
}

File: src/main/kotlin/net/atos/zac/app/zaak/converter/RESTZaaktypeRelatieConverter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaak.converter

import net.atos.client.zgw.ztc.model.generated.ZaakTypenRelatie
import net.atos.zac.app.zaak.model.RESTZaaktypeRelatie
import net.atos.zac.app.zaak.model.RelatieType
import net.atos.zac.util.UriUtil
import java.net.URI

fun convertToRESTZaaktypeRelatie(zaaktypUri: URI, relatieType: RelatieType) =
    RESTZaaktypeRelatie(
        zaaktypeUuid = UriUtil.uuidFromURI(zaaktypUri),
        relatieType = relatieType
    )

fun convertToRESTZaaktypeRelatie(zaakTypenRelatie: ZaakTypenRelatie) =
    RESTZaaktypeRelatie(
        zaaktypeUuid = UriUtil.uuidFromURI(zaakTypenRelatie.zaaktype),
        relatieType = RelatieType.valueOf(zaakTypenRelatie.aardRelatie.name)
    )
